provider/azure/config.go

```diff
@@ -65,6 +66,10 @@ func getConfig(configFile, subscriptionID, resourceGroup, userAssignedIdentityCl
 	if userAssignedIdentityClientID != "" {
 		cfg.UserAssignedIdentityID = userAssignedIdentityClientID
 	}
+	// If activeDirectoryAuthorityHost is provided explicitly, override existing one in config file
+	if activeDirectoryAuthorityHost != "" {
+		cfg.ActiveDirectoryAuthorityHost = activeDirectoryAuthorityHost
+	}
```

Comment:
One thing I thought about: what happens if `cloud` is set to a non-empty value (e.g. "AZURECHINACLOUD") but `activeDirectoryAuthorityHost` is also set to some non-empty value. I think we should return an error in that case